src/modules/settings/components/Window.jsx

```diff
@@ -0,0 +1,56 @@
+import React, {useEffect, useState} from 'react';
+import Modal from 'rsuite/lib/Modal/index.js';
+import Sidenav from './Sidenav.jsx';
+import About from '../pages/About.jsx';
+import ChatSettings from '../pages/ChatSettings.jsx';
+import DirectorySettings from '../pages/DirectorySettings.jsx';
+import ChannelSettings from '../pages/ChannelSettings.jsx';
+import Changelog from '../pages/Changelog.jsx';
+import ModalContext from '../context.js';
+
+function renderPage(page) {
+  switch (page) {
+    case 1:
+      return <ChatSettings />;
+    case 2:
+      return <DirectorySettings />;
+    case 3:
+      return <ChannelSettings />;
+    case 5:
+      return <Changelog />;
+    default:
+      return <About />;
+  }
+}
+
+function Window(props) {
+  const {open: current} = props;
+  const [page, setPage] = useState(1);
+  const [modal, setModal] = useState({open: current, enter: false});
+
+  useEffect(() => {
+    setModal({open: current, enter: false});
+  }, [props]);
+
+  const {open, enter} = modal;
+
+  return (
+    <ModalContext.Provider value={modal}>
+      <Modal
+        show={open}
+        onEntered={() =>
+          setTimeout(() => {
```

Comment:
The tables would fail to render properly on modal open, so I use this as a placeholder for when the modal is fully open and I found it would sometimes still fail and adding this extra delay would fix it.

Reply:
It would probably be smarter to have this delay in the actual Table.js file